src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,116 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.db.Storage.getPeople;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private StorageDao storageDao = new StorageDaoImpl();
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private Storage storage = new Storage();
+    private User bob;
+
+    @BeforeEach
+    void setUp() {
+        bob = new User("bob423", "123456", 19);
+    }
+
+    @AfterEach
+    void tearDown() {
+        new Storage();
+    }
+
+    @Test
+    void register_registerUsersByCorrectData_ok() {
+        storageDao.add(bob);
+        bob.setLogin("bob_terminator");
+        storageDao.add(bob);
+        bob.setLogin("bob_jenkins");
+        storageDao.add(bob);
+        int actual = getPeople().size();
+        int expected = 3;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_getUserByExistLogin_ok() {
```

Comment:
No need to test how our dao methods work. The task of the tests here is to cover all possible variants of correct and incorrect registration.